### PR TITLE
apparmor: add support for *-musl.

### PR DESCRIPTION
**Pull request by maxice8**
This adds apparmor support for musl arches and sets it to nocross instead.

cc @olivier-mauras @thypon 

it is untested.

### CODE REVIEW

File: srcpkgs/docker/template [hunk 1]
```diff
@@ -25,15 +25,8 @@ system_groups="docker"
 
 _docker_components="runc containerd tini proxy dockercli"
 
-build_options="apparmor"
```
Comment (Gottox) [inline srcpkgs/docker/template]: Is there a reason to remove the option instead of just default it to on?

**Comment by maxice8** [comment]
On Sun, Jul 08, 2018 at 12:47:36AM -0700, Enno Boland wrote:
> Gottox commented on this pull request.
> 
> 
> 
> > @@ -25,15 +25,8 @@ system_groups="docker"
>  
>  _docker_components="runc containerd tini proxy dockercli"
>  
> -build_options="apparmor"
> 
> Is there a reason to remove the option instead of just default it to on?
> 
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/711#pullrequestreview-135224752

i only added it because musl was being blocked from building docker because it didn't have apparmor, now that it has apparmor we can get rid of it.

**Comment by Gottox** [comment]
I don't see the need of ever removing buildoptions as long as they're supported by the package.

**Comment by maxice8** [comment]
I see no reason to keep this build_option, it completed it's mission of not keeping x86_64-musl stuck with old docker. There is also no reason to compile without apparmor support when it is available.

i'll keep this to the maintainer of the package to decide.

**Comment by Gottox** [comment]
It doesn't serve any purpose for us, like most `build_options` do. The idea is, that third parties can enable/disable features on a packages at will.

I might get docker to cross compile soon, so (at least I'm more optimistic about that than about apparmor) so the build flag remains useful for me too.

**Comment by Gottox** [comment]
Merge this for now. I'm delaying cross-build for docker till apparmor is fixed.

**Pull request closed without merging**